### ISSUE TITLE
extract css option not extracting?

### ISSUE THREAD

**Issue by dimitrioskarvounaris** [Nothing-Works/vuetifyjs-mix-extension#14]
I'm setting up currently a new starter project for Laravel with vuetify. Because of an issue with vuetify-loader 1.6 and Mix, and because it offers all these options, I opted to use your extension for simplicity. 

The project is mostly empty, while I'm trying to make the extract css function work. 

```
require('vuetifyjs-mix-extension');
const vuetifyOptions = {
    progressiveImages: true,
    extract: 'css/vuetify-components.css',
};
mix.vuetify('vuetify-loader', vuetifyOptions);
```

This however does not create another css file. I tried also the extract option with no other options, the result is the same. 

I have an extremely simple App right now, so I added another component to it, to make sure I have at least some vuetify components in it. Again, no extraction file was created.

I'm not sure if this is a bug or I'm doing something wrong. I studied all I could and found no explanation or solution.
Any ideas?

**Comment by Nothing-Works** [comment]
The code you have below should give you an error when you run `npm run dev`, as you are not even compiling your `js` files and you are asking the extension to include `vuetify-loader` so it does not know what to do.

```js
require('vuetifyjs-mix-extension');
const vuetifyOptions = {
    progressiveImages: true,
    extract: 'css/vuetify-components.css',
};
mix.vuetify('vuetify-loader', vuetifyOptions);
```

I am using this feature and it works for me, you may consider the code below.

```js
//first require mix
const mix = require('laravel-mix');

require('vuetifyjs-mix-extension');

//you need to compile you js file, the css or sass is optional and then use vuetify.
mix.js('resources/js/app.js', 'public/js')
   .sass('resources/sass/app.scss', 'public/css')
   .vuetify('vuetify-loader', {
    extract: 'css/vuetify-components.css',
   });
```

**Comment by dimitrioskarvounaris** [comment]
Of course I'm compiling the files, I just left the "default" lines out, and copied only the code related to the extension. 

I'll try again, removing some other options I have like for sourcemaps, only with the code you posted. And have a look on your repo and see if it works.

**Comment by dimitrioskarvounaris** [comment]
So, after retrying and still not working, I replaced some files in my project fully with these from the repo: `webpack.mix.js`, `app.js`, `plugin.js`, `ExampleComponent.vue` (I think those are all who seem to matter), so that I have a "clean" starting point and be certain, that there is nothing that I missed and may be different. 

Yet, I still get no extracted css, `yarn dev` gives the following:

```
 DONE  Compiled successfully in 10821ms                                                                                    13:12:06

                                                                               Asset      Size   Chunks             Chunk Names
                                                                        /css/app.css   765 KiB  /js/app  [emitted]  /js/app
                                                                          /js/app.js  2.59 MiB  /js/app  [emitted]  /js/app
  fonts/vendor/@mdi/materialdesignicons-webfont.eot?4a837d054b5f2a37170df8a275a13816   919 KiB           [emitted]
  fonts/vendor/@mdi/materialdesignicons-webfont.ttf?b0fd91bb29dcb296a9a37f8bda0a2d85   918 KiB           [emitted]
fonts/vendor/@mdi/materialdesignicons-webfont.woff2?f1997a8aba8a498fe4032e3b56e871ca   296 KiB           [emitted]
 fonts/vendor/@mdi/materialdesignicons-webfont.woff?31d47085569e772c0f57aa8ec8381a5c   421 KiB           [emitted]
Done in 12.74s.
```

So, as a last, I cloned the other repo and run `yarn install` and  `yarn dev`.
I get the following:
```
 DONE  Compiled successfully in 8102ms                                                                                     13:25:22

       Asset      Size   Chunks             Chunk Names
/css/app.css   0 bytes  /js/app  [emitted]  /js/app
  /js/app.js  2.59 MiB  /js/app  [emitted]  /js/app
Done in 10.33s.
```

I can only imagine, there is something with the tools maybe different... like different nodejs version?! I'm not sure, what it is, but I'm certain, that it's not working in my case at all. Any more ideas?

**Comment by dimitrioskarvounaris** [comment]
Ok, so, out of a thought, I tried `npm run dev` intead of `yarn dev`.

On the same installation, with node_modules installed with yarn, as well on the repo you linked as also on my own without any of the previously tried modifications, `npm run dev` DOES extract the css/vuetify-component.css, while `yarn dev` DOES NOT. This is 100% for certain, absolutely no mistake here. I'm certain, you could reproduce it. 

But how can that be? There shouldn't be normally a difference between these two, as they are only executing tasks and not doing it themselves?

**Comment by dimitrioskarvounaris** [comment]
Since I don't know if this is an issue with this extension or rather one of Yarn, I submitted https://github.com/yarnpkg/yarn/issues/8319

If you have though any idea what it could be, I would really appreciate your input.

**Comment by Nothing-Works** [comment]
That's very surprising, I did not try `yarn` before. I will do it soon.

**Comment by Nothing-Works** [comment]
I am sorry I still can not reproduce this. It still works for me I can still get the extracted `css` with my example repo. I will send my system info soon.

**Comment by Nothing-Works** [comment]
```bash
$ node -v
v14.8.0
```

```bash
$ npm -v
6.14.8
```

```bash
$ yarn --version
1.22.4
```

```bash
Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19041
 OS Platform: Windows
 RID:         win10-x64
```

**Comment by dimitrioskarvounaris** [comment]
The only difference I see here is that I'm using node v12, which is also the recommended version. I downgraded it some time ago, because there was something in a project, that was requiring to use it and were not working with the latest. I don't recall though what exactly it was. Otherwise, our environment is identical.

**Comment by dimitrioskarvounaris** [comment]
No, that's not it. I installed nvm to have multiple node versions and switched to the same like yours. Yarn is still causing this file to be missed, while npm works. 
Something else is responsible for this issue, but seems we can't get to ground. *sigh*

**Comment by dimitrioskarvounaris** [comment]
So, I tried it on the laptop and I had the same result at first. But, I could nail down the culprit even more. When I run it in Git Bash as shell, then the error occurs. When I run it on Windows command shell, it works with yarn too! 
So, the issue is related to the shell used, which is quite weird and I can't think why that could be. May be related to one of the included modules down the line, which may have some issue with paths or something like this.

**Comment by Nothing-Works** [comment]
I do not have any problems at all, I can not debug this issue. I am going to close this if you do not mind. Feel free to provide more info if you find what's causing it.

**Comment by dimitrioskarvounaris** [comment]
There are differences in the configuration between the two executions. I checked the "config" object. I also noticed, that it seems to not pick up the variables files in the one with the issue, too.

![image](https://user-images.githubusercontent.com/66212726/91292983-9d199980-e797-11ea-8420-565b54c6fdad.png)

Notice the difference on how it has the path wrongly coded. Seems like one of the typical windows/linux path issues. Still it is weird, that it does not happen with npm and only yarn, that's something I cannot figure out why.

**Comment by Nothing-Works** [comment]
I also use git bash on windows.

**Comment by dimitrioskarvounaris** [comment]
And you do not have this issue with git bash+yarn? 
It is most weird that it's the only combination where it happens, on my laptop too where I had to install node/npm/yarn fresh since I had them not on it?

**Comment by Nothing-Works** [comment]
![Capture](https://user-images.githubusercontent.com/18606648/91294088-1becf180-e7ed-11ea-8b35-5cebc75df2e3.PNG)

**Comment by dimitrioskarvounaris** [comment]
Thanks. I just upgraded to it too, but no difference. I had the previous one. *sigh* So frustrating... I'll search a bit more, maybe I find the culprit. Thanks for your help.

**Comment by Nothing-Works** [comment]
Yes, this is very frustrating, especially when it comes to the path problem. Please let me know if I can do something for my repo. Thanks.

**Comment by dimitrioskarvounaris** [comment]
I figured it out!!!

Check the short video, that shows the results and how a simple change in your repo fixes it. 


The problem? node and npm/yarn are all setting some environment variables when they run. The PWD variables does not exist in the windows cmd and powershell. It generally exists only linux environments. Git Bash though usually seems to set it (there may be exceptions, I don't know), as it emulates the linux behaviour partially. Yarn does delegate that environment variable further, while npm seems to not or remove it, when run through a shell under windows. I suppose, npm does not recognize it's a windows shell to remove it when it's run from within the scripts. That would explain, why the `npm run development` executed from the "dev" scripts run command plays no role, when it's initiated with `yarn run dev`. 

**tldr;** Having the PWD variable in windows environment will set the path in linux format, but resolving it, does treat it in node as a windows path format and so we get what can be seen in the screenshot further above, a mixed path, which is incorrect and as such it cannot resolve the vuetify path. Such, no extracted vuetify css and no sass variables.

Solution: Either remove in the file `src/getPath.php` querying `process.env.PWD` completely, or if it's somewhere else required, check if it's a windows environment and skip assigning it on windows.

**Comment by dimitrioskarvounaris** [comment]
@Nothing-Works Did you see my last comment? (not sure if closed issues receive notifications)

**Comment by Nothing-Works** [comment]
yes, I saw your comment, sorry I need more testing for that.
